silent; code_execution_config; exit; version

sonichi opened the pull request:
<!-- Thank you for your contribution! Please review https://microsoft.github.io/FLAML/docs/Contribute before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

1. Support silent mode to hide unnecessary messages in chess and group chat. (experimental)
2. Support lookback for multiple messages in code_execution_config. (experimental) @LittleLittleCloud 
3. Improve group chat speaker selection. @LittleLittleCloud 
4. Support exit group chat when human input is "exit".
5. Bump version to 2.0.0.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

<!-- - I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks). -->
- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

qingyun-wu reviewed (commented):
Did you compile and check if the doc pages on the website are good? If so, LGTM.

sonichi commented:
> Did you compile and check if the doc pages on the website are good? If so, LGTM.

Partially. I made more changes when I checked. But it'll be good to have a second eye on it.

thinkall commented:
Could you also help fix a few typos in `autogen_agentchat_RetrieveChat.ipynb`?

```
-    "FLAML requires `Python>=3.8`. To run this notebook example, please install flaml with the [mathchat] option.\n",
+    "FLAML requires `Python>=3.8`. To run this notebook example, please install flaml with the [retrievechat] option.\n",


-    "We start by initialzing the `RetrieveAssistantAgent` and `RetrieveUserProxyAgent`. The system message needs to be set to \"You are a help
ful assistant.\" for RetrieveAssistantAgent. The detailed instructions are given in the user message. Later we will use the `RetrieveUserProxyA
gent.generate_init_prompt` to combine the instructions and a math problem for an initial prompt to be sent to the LLM assistant."
+    "We start by initialzing the `RetrieveAssistantAgent` and `RetrieveUserProxyAgent`. The system message needs to be set to \"You are a help
ful assistant.\" for RetrieveAssistantAgent. The detailed instructions are given in the user message. Later we will use the `RetrieveUserProxyA
gent.generate_init_prompt` to combine the instructions and a retrieval augmented generation task for an initial prompt to be sent to the LLM as
sistant."
```

thinkall commented on notebook/autogen_agentchat_RetrieveChat.ipynb:
`pip install flaml[retrievechat]~=2.0.0rc5` also needs to be updated.

thinkall reviewed (changes requested):
I tried to run `autogen_agentchat_RetrieveChat.ipynb`, it failed with silent=True.

thinkall reviewed (commented):
I've fixed errors in retrieve chat. One more thing is that there is a function `_generate_math_reply` in `math_user_proxy_agent.py`, but I don't see it being called when running `autogen_agentchat_MathChat.ipynb`. If it will be called, looks like it should be updated to handle parameter `config`.

sonichi commented:
> I've fixed errors in retrieve chat. One more thing is that there is a function `_generate_math_reply` in `math_user_proxy_agent.py`, but I don't see it being called when running `autogen_agentchat_MathChat.ipynb`. If it will be called, looks like it should be updated to handle parameter `config`.

Thanks. I fixed that.
